modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -785,11 +785,11 @@ OPENCV_HAL_IMPL_NEON_SHIFT_OP(v_int64x2, s64, int64, s64)
 template<int n> inline _Tpvec v_rotate_right(const _Tpvec& a) \
 { return _Tpvec(vextq_##suffix(a.val, vdupq_n_##suffix(0), n)); } \
 template<int n> inline _Tpvec v_rotate_left(const _Tpvec& a) \
-{ return _Tpvec(vextq_##suffix(vdupq_n_##suffix(0), a.val, _Tpvec::nlanes - n)); } \
+{ return _Tpvec(vextq_##suffix(vdupq_n_##suffix(0), a.val, (_Tpvec::nlanes - 1) - n)); } \
 template<int n> inline _Tpvec v_rotate_right(const _Tpvec& a, const _Tpvec& b) \
 { return _Tpvec(vextq_##suffix(a.val, b.val, n)); } \
 template<int n> inline _Tpvec v_rotate_left(const _Tpvec& a, const _Tpvec& b) \
-{ return _Tpvec(vextq_##suffix(b.val, a.val, _Tpvec::nlanes - n)); }
+{ return _Tpvec(vextq_##suffix(b.val, a.val, (_Tpvec::nlanes - 1) - n)); }
```

Comment:
It looks like this will change the result. `v_rotate_left<_Tpvec::nlanes - 1>` will produce `b.val`, while it should combine parts of `a.val` and `b.val`

Reply:
Right, let me get back with tested result.
I only confirmed that it let the build go through.

Reply:
Just try to add a special case if immediate constant value equal zero.
````
if (n == 0) return a;
````